### PR TITLE
Drawer: Introduce a size property that set's width percentage and minWidth 

### LINKED ISSUE #27426: Inspect drawer tabs break on very narrow screens
**What happened**:
For narrow screens the labels of the inspect drawer tabs get cut off. They probably should change to a dropdown like the panel edit sidebar does.

image

### PR DESCRIPTION
**Pull request by torkelo**
Fixes #27426
 
Trying to solve the problem of our drawer that scale poorly on smaller screens as we only set a fixed percentage width right now, and 40% or 50% of a small screen is way to small for many of our drawers. 

* Add size prop with values `sm` | `md` | `lg`  and deprecate width prop 
* size prop sets percentage-based width AND min-width 
* Adds media query to improve handling large drawer with high min-width on smaller screens
* Adjust media query to leave a small margin where you can still see background to make it more clear you are in a drawer even on very small screens (before it was 100%). 
* Deprecates inline drawers that have been broken for 2 years (https://github.com/grafana/grafana/issues/39390)
* Update Storybook & mdx 

Open question:

* Maybe with this we can remove the "expandable" feature? (the chevron which when clicked expands the drawer to 100%). That interaction could be a bit unnecessary after this PR and would simplify the drawer header. 

[New-Features-in-v7-4---gdev-dashboards---Dashboards---Grafana.webm](https://user-images.githubusercontent.com/10999/236156161-7d41fcc8-b719-463b-9511-1fe42cf9916f.webm)

Example of a problem drawer:
Screenshot from 2023-05-04 10-52-45

### CODE REVIEW

**Review by amy-super — APPROVED** [review]
I think these changes work really well. In my head there are two outstanding questions - one that you raise and one that I thought of. 

1. Do we eliminate the expander for the drawer? 
The only place I can think of that really benefits from that is Dashboard Insights - and even then, a large drawer seems to be plenty on the examples I've found. So I vote yes, but I'd want to poll the rest of the UX team to make sure we're not breaking anything for them. 

2. This one has been on my list for awhile, but it bugs me that the megamenu opens under topnav but drawers open over it. They are for different things, and I think there is an argument to be made that it's OK as is. We want to not totally hide the nav elements when you are in megamenu, but drawers benefit from the extra height given their big headers. So I think it's OK? But wanted to raise it since we're talking drawers. Feel free to ignore this point - I have it tracked in #55623.

**Comment by torkelo** [comment]
@amy-super for point nr 2 (position under nav) I raised a follow-up PR https://github.com/grafana/grafana/pull/67824 , looks promising.

File: packages/grafana-ui/src/components/Drawer/Drawer.tsx [hunk 1]
```diff
@@ -66,31 +70,30 @@ export function Drawer({
...
   }, []);
 
-  const content = <div className={drawerStyles.content}>{children}</div>;
-  const style: CSSProperties = {};
-  if (inline) {
-    style.position = 'absolute';
-  }
+  // deprecated width width prop now defaults to empty string which make the size prop take over
+  const fixedWidth = isExpanded ? '100%' : width ?? '';
+  const useSizeWidth = !fixedWidth && !isExpanded;
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/Drawer/Drawer.tsx:75]: The only way for `fixedWidth` to be `falsy` is if `isExpanded` is also `false`, so I think `useSizeWidth = !fixedWidth` is enough no?

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Love how this makes it easier to have different sizes of `Drawer` instead of just an arbitrary `width` 🙌

**Comment by JoaoSilvaGrafana** [comment]
> @JoaoSilvaGrafana inline supports seem to have been broken for quite some time, #39390
> 
> Wonder if we should deprecate the prop instead of fixing it. Can't think of where we would use inline drawers. Maybe better to add that support when we really need it

@torkelo I agree that if it is not used and it is broken better to just deprecate it... and then if we need this solution again we revisit it 👍

File: packages/grafana-ui/src/components/Drawer/Drawer.tsx (hunk 1 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/Drawer/Drawer.tsx]: good catch :) fixed!

**Comment by torkelo** [comment]
@JoaoSilvaGrafana deprecated inline part (added deprecated annotation to prop, and removed story). Also update the mdx to include info on the sizes. Think this is ready now

**Review by JoaoSilvaGrafana — APPROVED** [review]
Lgtm!

**Pull request merged**